htlcswitch/interfaces.go

```diff
@@ -135,7 +135,21 @@ type ChannelUpdateHandler interface {
 	// ShutdownIfChannelClean shuts the link down if the channel state is
 	// clean. This can be used with dynamic commitment negotiation or coop
 	// close negotiation which require a clean channel state.
-	ShutdownIfChannelClean() error
+	ShutdownHtlcManager()
```

Comment:
Note for reviewers: I am having second thoughts about this as I think that having this kind of "unsafe" method on the interface isn't a good idea. I think it may be better to replace it with a "true" shutdown method that implicitly calls the flush before calling an internal unsafe `shutdownHtlcManager`